plugins/announcements/src/components/AnnouncementsTimeline/AnnouncementsTimeline.tsx

```diff
@@ -0,0 +1,138 @@
+import { useApi, useRouteRef } from '@backstage/core-plugin-api';
+import { Typography, Box } from '@material-ui/core';
+import {
+  Timeline,
+  TimelineItem,
+  TimelineContent,
+  TimelineOppositeContent,
+  TimelineConnector,
+  TimelineDot,
+  TimelineSeparator,
+} from '@material-ui/lab';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { Announcement, announcementsApiRef } from '../../api';
+import Stack from '@mui/material/Stack';
+import { DateTime } from 'luxon';
+import { announcementViewRouteRef } from '../../routes';
+
+/**
+ * Props for the AnnouncementsTimeline component.
+ */
+export type AnnouncementsTimelineProps = {
+  /**
+   * Options for configuring the results of the timeline.
+   */
+  options?: {
+    /**
+     * Options for configuring the number of results to display.
+     */
+    results?: {
+      /**
+       * The maximum number of results to display.
+       */
+      max?: number;
+    };
+    /**
+     * Options for configuring the alignment and width of the timeline.
+     */
+    timeline?: {
+      /**
+       * The alignment of the timeline items. Can be 'left', 'right', or 'alternate'.
+       */
+      align?: 'left' | 'right' | 'alternate';
+      /**
+       * The minimum width of the timeline.
+       */
+      minWidth?: string;
+    };
+  };
+};
+
+/**
+ * Default alignment for the timeline.
+ */
+const DEFAULT_TIMELINE_ALIGNMENT = 'alternate';
+
+/**
+ * Default width for the timeline.
+ */
+const DEFAULT_TIMELINE_WIDTH = '425px';
+
+/**
+ * Default maximum number of results to display.
+ */
+const DEFAULT_RESULTS_MAX = 10;
+
+/**
+ * Renders a timeline of announcements.
+ *
+ * @param options - The options for the announcements timeline.
+ * @returns The rendered announcements timeline.
+ */
+export const AnnouncementsTimeline = ({
+  options,
+}: AnnouncementsTimelineProps) => {
+  const [announcements, setAnnouncements] = useState<Announcement[]>([]);
+  const announcementsApi = useApi(announcementsApiRef);
+  const viewAnnouncementLink = useRouteRef(announcementViewRouteRef);
+
+  useEffect(() => {
+    async function fetchData() {
+      const { results } = await announcementsApi.announcements({
+        max: options?.results?.max ?? DEFAULT_RESULTS_MAX,
+      });
+      setAnnouncements(results);
+    }
+
+    fetchData();
+  }, [announcements, announcementsApi, options?.results?.max]);
```

Comment:
It seems that the component is constantly fetching data because of `useEffect` hook dependency on `announcements`.  I noticed that because the component was hitting the API  backend every ~1.5ms. 

I tried this approach to fetch data only when component is mounted. Maybe there is a better way to achieve this 😄 
```suggestion
  useEffect(() => {
    let isMounted = true;

    async function fetchData() {
      try {
        const { results } = await announcementsApi.announcements({
          max: options?.results?.max ?? DEFAULT_RESULTS_MAX,
        });

        if (isMounted) {
          setAnnouncements(results);
        }
      } catch (error) {
        // Handle error
      }
    }

    fetchData();

    return () => {
      isMounted = false;
    };
  }, [announcementsApi, options?.results?.max]);
```

Reply:
Hey thanks for the info. I should have time to look into this over the weekend.

Reply:
@gaelgoth, I'm not sure the `useEffect` hook needs to depend on `announcements`. You get the same result when removing it as a dep.